File: logisticos_app/lib/models/noticia.dart
class Noticia {
  final int id;
  final String autorNombre;
  final String titulo;
  final String contenido;
  final String? planta;
  final bool destacada;
  final String fechaPublicacion;
  final String? imagenPresignedUrl;
  final String? createdAt;

  const Noticia({
    required this.id,
    required this.autorNombre,
    required this.titulo,
    required this.contenido,
    this.planta,
    required this.destacada,
    required this.fechaPublicacion,
    this.imagenPresignedUrl,
    this.createdAt,
  });

  factory Noticia.fromJson(Map<String, dynamic> json) {
    return Noticia(
      id: json['id'],
      autorNombre: json['autorNombre'] ?? '',
      titulo: json['titulo'] ?? '',
      contenido: json['contenido'] ?? '',
      planta: json['planta'],
      destacada: json['destacada'] ?? false,
      fechaPublicacion: json['fechaPublicacion'] ?? '',
      imagenPresignedUrl: json['imagenPresignedUrl'],
      createdAt: json['createdAt'],
    );
  }
}

File: CoorpoSanpedroBackend/src/main/java/com/festival/entity/Noticia.java
package com.festival.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "noticias")
@Getter
@Setter
@NoArgsConstructor
public class Noticia extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "autor_id", nullable = false)
    private Usuario autor;

    @Column(nullable = false, length = 300)
    private String titulo;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String contenido;

    @Column(length = 100)
    private String planta;

    @Column(nullable = false)
    private boolean destacada = false;

    @Column(name = "imagen_ruta")
    private String imagenRuta;

    @Column(name = "fecha_publicacion", nullable = false)
    private String fechaPublicacion;
}
